doc/scripts/examples.pl

```diff
@@ -0,0 +1,229 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2013 - 2019 by the deal.II authors
+##
+## The original file is from the deal.II library. It has been modified for
+## Ginkgo.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE.md at
+## the top level directory of deal.II.
+##
+## ---------------------------------------------------------------------
+
+use strict;
+
+my $example_file = shift;
+open EXAMPLE, "<$example_file";
+
+# Print the first part of example.hpp.in up until the point where we
+# find the line with '@@EXAMPLE_MAP@@'
+while (my $line = <EXAMPLE>)
+{
+  last if($line =~ m/\@\@EXAMPLE_MAP\@\@/);
+  print $line;
+}
+
+# List of additional node attributes to highlight purpose and state of the example
+my %style = (
+    "basic"          => ',height=.8,width=.8,shape="octagon",fillcolor="green"',
+    "techniques"     => ',height=.35,width=.35,fillcolor="orange"',
+    "logging"         => ',height=.25,width=.25,fillcolor="yellow"',
+    "stopping-criteria"         => ',height=.25,width=.25,fillcolor="lightblue"',
+    "preconditioners" => ',height=.25,width=.25,fillcolor="blue"',
+    "unfinished"     => ',height=.25,width=.25,style="dashed"',
+    );
+
+# Print a preamble setting common attributes
+print << 'EOT'
+digraph ExamplesMap
+{
+  overlap=false;
+  edge [fontname="FreeSans",
+        fontsize="10",
+        labelfontname="FreeSans",
+        labelfontsize="10",
+        color="black",
+        style="solid"];
+  node [fontname="FreeSans",
+        fontsize="10",
+        shape="rectangle",
+        height=0.2,
+        width=0.4,
+        color="black",
+        fillcolor="white",
+        style="filled"];
+EOT
+    ;
+
+# Print all nodes of the graph by looping over the remaining
+# command line arguments denoting the example programs
+
+my $example;
+foreach $example (@ARGV)
+{
+    # read first line of tooltip file
+    open TF, "$example/doc/tooltip"
+        or die "Can't open tooltip file $example/doc/tooltip";
+    my $tooltip = <TF>;
+    close TF;
+    chop $tooltip;
+
+    # read first line of 'kind' file if it is a example;
+    # otherwise assume it is a code gallery program. for
+    # each of them, output something for 'dot' to generate
+    # the dependencies graph from
+    if ($example =~ /example-/
```

Comment:
I do not really see why we would have to use the form `example-<number>` according to this here. Though admittedly I did not go through the scripts in detail. It looks like the examples are passed by argument to the script anyway, could we not pass a list of everything in ginkgo/examples and be done with it? We could even sort the examples and give a number to each of them, if a number is required...

Reply:
Yes, sure. We can fix this together.